### PR TITLE
simple datastore sample app

### LINKED ISSUE #1291: Create a simple Datastore sample app

### PR DESCRIPTION
**Pull request by dmitry-s**
fixes #1291

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/pom.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-gcp-starter-data-datastore</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-rest</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.shell</groupId>
+            <artifactId>spring-shell-starter</artifactId>
```
[the remark this answers is not in the document]
Comment (dmitry-s) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/pom.xml]: This application creates an interactive shell, that's why we need `spring-shell`

[the remark this answers is not in the document]
Comment (artembilan) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/pom.xml]: OK. But why? Isn't REST not enough for us to demonstrate a feature simple way?
What is the benefit of that shell for this particular sample?

Thanks

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/test/java/com/example/DatastoreBookshelfExampleTests.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * @author Dmitry Solomakha
+ */
+
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/test/java/com/example/DatastoreBookshelfExampleTests.java]: No need for a blank line here.
```suggestion
```

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/test/java/com/example/DatastoreBookshelfExampleTests.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	@BeforeClass
+	public static void checkToRun() {
+		assumeThat(
+				"Datastore-sample integration tests are disabled. Please use '-Dit.datastore=true' "
+						+ "to enable them. ",
+				System.getProperty("it.datastore"), is("true"));
+	}
+
+	@Test
+	public void testSaveBook() {
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/test/java/com/example/DatastoreBookshelfExampleTests.java]: Can we add tests for all of the other actions in the app?

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitry-s) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/pom.xml]: Using shell makes it more interactive. Users don't have to use curl and format their requests as json.

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/main/java/com/example/DatastoreBookshelfExample.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+
+/**
+ * @author Dmitry Solomakha
+ */
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-basic-sample/src/main/java/com/example/DatastoreBookshelfExample.java]: I think maybe a small little java doc on each of the 3 classes would be helpful. 

something short like 
Book.java  "this represents a single book stored in Datastore"
DatastoreBookshelfExample.java "this contains the main method and performs common operations with books"
BookRepository.java "this interface contains custom-defined query methods of which implementations are generated for you"

**Pull request merged**